sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -597,4 +597,49 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
         })
     })
   }
+
+  test("inner join respects collations") {
```

Comment:
How is this test different from golden file test?

Reply:
It's same. Having this test may come in handy while developing hash based joins, given that it's easier to debug functional tests then golden files.

But if you feel that it is redundant I can remove it.

Reply:
done.